crates/curp/src/server/raw_curp/log.rs

```diff
@@ -111,21 +118,98 @@ impl<C: Command> LogEntryVecDeque<C> {
 
     /// push a log entry into the back of queue
     fn push_back(&mut self, entry: Arc<LogEntry<C>>) -> Result<(), bincode::Error> {
+        #![allow(clippy::indexing_slicing)]
         let entry_size = serialized_size(&entry)?;
 
+        if entry_size > self.batch_limit {
+            warn!("entry_size of an entry > batch_limit, which may be too small.",);
+        }
+
         self.entries.push_back(entry);
-        let Some(&pre_entries_size) = self.batch_index.back() else {
-            unreachable!("batch_index cannot be None")
-        };
-        self.batch_index
-            .push_back(pre_entries_size.overflow_add(entry_size));
+        self.entry_size.push_back(entry_size);
+        self.batch_index.push_back(0); // placeholder
+
+        if entry_size > self.batch_limit {
+            self.handle_oversized_entry();
+            return Ok(());
+        }
+
+        while self.last_batch_size + entry_size > self.batch_limit
```

Comment:
In what situation will this `self.first_entry_at_last_batch >= self.entries.len()` come true?